lib/Serialization/Deserialization.cpp

```diff
@@ -3536,8 +3546,8 @@ class swift::DeclDeserializer {
 
     theEnum->computeType();
 
-    handleInherited(theEnum,
-                    rawInheritedAndDependencyIDs.slice(0, numInheritedTypes));
+    auto rawInheritedIDs = rawInheritedAndDependencyIDs.slice(0, numInherited);
+    handleInherited(theEnum, rawInheritedIDs);
```

Comment:
I just merged the class and struct versions for this, so you can switch to it if you want.